src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,40 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useAppDispatch } from '../../app/hooks';
+import { actions as filtrationAction } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const [selectedOption, setSelectedOption] = useState('all');
+  const [input, setInput] = useState('');
+  const dispatch = useAppDispatch();
+
+  const queryHandler = () => {
+    dispatch(filtrationAction.setQuery(input, selectedOption));
+  };
+
+  const selectHandler = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedOption(event.target.value);
+  };
+
+  const inputHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInput(event.target.value);
+  };
+
+  const clearInputButton = () => {
+    setInput('');
+  };
+
+  useEffect(() => {
+    queryHandler();
+  }, [input, selectHandler]);
```

Comment:
Why do you need the selectHandler in the dependency array? Fix it, please.
```suggestion
  useEffect(() => {
    queryHandler();
  }, [queryHandler]);
```